### PR TITLE
Only allow api.data.gov to access the API

### PR DESCRIPTION
**Pull request by jadudm**
This branch:

1. Introduces terraform config for PostgREST JWT secrets
2. Requires JWT secrets from `api.data.gov` in order to resolve API requests
3. Has tooling (Django commands) for creating JWT secrets

We have made the corresponding changes in the `api.data.gov` backend, and this should, therefore, make `api.fac.gov` available for public use. It should also mean that `fac-prod-postgrest.app.cloud.gov` will no longer accept API requests *except* from `api.data.gov`.

Secrets were added to `dev`, `staging`, ``preview` and `prod` in Github for workflows/deploys.

### CODE REVIEW

File: backend/dissemination/management/commands/create_api_jwt.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+from math import floor
+
+import jwt
+import sys
+
+class Command(BaseCommand):
+    help = """
+    Runs sql scripts  to recreate views for the postgrest API.
```
Comment (mogul) [inline backend/dissemination/management/commands/create_api_jwt.py:11]:
```suggestion
    Creates JWT tokens for use in authenticating requests to PostgREST
```

File: terraform/shared/modules/env/postgrest.tf [hunk 2]
```diff
@@ -33,7 +33,8 @@ resource "cloudfoundry_app" "postgrest" {
   environment = {
     PGRST_DB_URI : cloudfoundry_service_key.postgrest.credentials.uri
     PGRST_DB_SCHEMAS : "api_v1_0_0_beta"
-    PGRST_DB_ANON_ROLE : "anon"
+    PGRST_DB_ANON_ROLE : "anon",
```
Comment (mogul) [inline terraform/shared/modules/env/postgrest.tf:36]:
```suggestion
    PGRST_DB_ANON_ROLE : "anon"
```

File: terraform/shared/modules/env/postgrest.tf [hunk 3]
```diff
@@ -33,7 +33,8 @@ resource "cloudfoundry_app" "postgrest" {
...
+    PGRST_JWT_SECRET: var.pgrst_jwt_secret
```
Comment (mogul) [inline terraform/shared/modules/env/postgrest.tf:37]:
```suggestion
    PGRST_JWT_SECRET : var.pgrst_jwt_secret
```

Comment (mogul) [reply terraform/shared/modules/env/postgrest.tf:37]: Fixes fmt failure

**Review by mogul — CHANGES_REQUESTED** [review]

File: backend/.profile
Comment (jadudm) [inline backend/.profile:1]: Need to merge `main` again.

File: backend/dissemination/README.md [hunk 4]
```diff
@@ -1,12 +1,95 @@
...
+
+# Creating a JWT secret
+
+The command
+
+```
+fac create_api_jwt <role> <passphrase>
+```
+
+creates a JWT secret. The passphrase must be [at least 32 characters long](https://postgrest.org/en/v10.2/tutorials/tut1.html#:~:text=32%20characters%20long.-,Note,-Unix%20tools%20can).
```
Comment (jadudm) [inline backend/dissemination/README.md:17]: Add a UNIX one-liner for generating a reasonable secret for feeding into this.

**Comment by danswick** [comment]
@jadudm per discussion, our test environment doesn't receive the JWT secret like the other environments do, which causes docker to substitute an empty string. I added a fallback value to `docker-compose-web`, which should only ever run in a GH actions test environment. Also did a little YAML standardization. See [this set of commits](https://github.com/GSA-TTS/FAC/pull/1663/files/e46e1824e485712991bc908e3a795c55f955ae5b..08ee40a181988a039eb5cbf4b7c5e280962dab72) for my changes.

**Review by mogul — APPROVED** [review]

**Pull request merged**